tenets/codelingo/go/writes-to-async-pointer/example.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+)
+
+type thing struct {
+	name string
+}
+
+func getNameFromPointer(t *thing) {
+	fmt.Println(t.name)
+}
+
+func getName(t thing) {
+	fmt.Println(t.name)
+}
+
+func setNameOnPointer(t *thing) {
+	t.name = "car"
+}
+
+func newThingPointer(name string) *thing {
+	return &thing{name}
+}
+
+func newThing(name string) thing {
+	return thing{name}
+}
+
+func main() {
```

Comment:
is there a reason we usually stick all the issues in a single `main` func? One function per issue would be a bit more readable – or at least breaking out the `t2` case to demonstrate that we don't raise an issue when there are no writes happening.

Also, documentation on the components would be really helpful – I look to the examples first to understand the tenet. I'll add some vague suggestions.